src/main/java/io/appium/java_client/android/AndroidMobileOptions.java

```diff
@@ -0,0 +1,1716 @@
+    }
+
+    /**
+     * Set the App to skip capturing logcat.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#SKIP_LOGCAT_CAPTURE
+     */
+    public AndroidMobileOptions setSkipLogcatCapture() {
+        return setSkipLogcatCapture(true);
+    }
+
+    /**
+     * Set whether to skip capturing logcat.
+     *
+     * @param bool when true does not capture logcat.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#SKIP_LOGCAT_CAPTURE
+     */
+    public AndroidMobileOptions setSkipLogcatCapture(boolean bool) {
+        return amend(AndroidMobileCapabilityType.SKIP_LOGCAT_CAPTURE, bool);
+    }
+
+    /**
+     * Get whether to skip capturing logcat.
+     *
+     * @return true if not capturing logcat.
+     * @see AndroidMobileCapabilityType#SKIP_LOGCAT_CAPTURE
+     */
+    public boolean doesSkipLogcatCapture() {
+        return (boolean) getCapability(AndroidMobileCapabilityType.SKIP_LOGCAT_CAPTURE);
+    }
+
+    /**
+     * Set the app to skip unlock during session creation.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#SKIP_UNLOCK
+     */
+    public AndroidMobileOptions setSkipUnlock() {
+        return setSkipUnlock(true);
+    }
+
+    /**
+     * Set whether to skip unlock during session creation.
+     *
+     * @param bool when true skips unlock during session creation.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#SKIP_UNLOCK
+     */
+    public AndroidMobileOptions setSkipUnlock(boolean bool) {
+        return amend(AndroidMobileCapabilityType.SKIP_UNLOCK, bool);
+    }
+
+    /**
+     * Get whether to skip unlock during session creation.
+     *
+     * @return true if unlock during session creation is skipped.
+     * @see AndroidMobileCapabilityType#SKIP_UNLOCK
+     */
+    public boolean doesSkipUnlock() {
+        return (boolean) getCapability(AndroidMobileCapabilityType.SKIP_UNLOCK);
+    }
+
+    /**
+     * Set the port used to connect to the driver.
+     *
+     * @param port is the port used to connect to the driver.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#SYSTEM_PORT
+     */
+    public AndroidMobileOptions setSystemPort(String port) {
+        return amend(AndroidMobileCapabilityType.SYSTEM_PORT, port);
+    }
+
+    /**
+     * Get the port used to connect to the driver.
+     *
+     * @return String of the port used to connect to the driver.
+     * @see AndroidMobileCapabilityType#SYSTEM_PORT
+     */
+    public String getSystemPort() {
+        return (String) getCapability(AndroidMobileCapabilityType.SYSTEM_PORT);
+    }
+
+    /**
+     * Set the app to enable Unicode input.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#UNICODE_KEYBOARD
+     */
+    public AndroidMobileOptions setUnicodeKeyboard() {
```

Comment:
I would not include this cap at all. This only has effect for UiAutomator1 backend, which is considered obsolete

Reply:
Makes sense. Can we add that knowledge to the documentation and/or CapabilityType file?